Dockerfile

```
@@ -380,7 +380,10 @@ RUN apt-get install -y curl
 RUN mkdir -p /dart-protobuf
 ARG PROTOC_GEN_DART_VERSION
 RUN curl -sSL https://api.github.com/repos/google/protobuf.dart/tarball/protoc_plugin-${PROTOC_GEN_DART_VERSION} | tar xz --strip 1 -C /dart-protobuf
-WORKDIR /dart-protobuf/protoc_plugin 
+WORKDIR /dart-protobuf/protoc_plugin
+# Use Dart mirror to work around connectivity problems to default host when building in QEMU
+# https://stackoverflow.com/questions/70729747
+ARG PUB_HOSTED_URL=https://pub.flutter-io.cn
```

```suggestion
ARG PUB_HOSTED_URL=https://pub.dev
```
Doesn't this one work?

---

https://pub.dev does not work under certain network conditions due to how the remote server is configured, the error is `Connection closed before full header was received.`

Other users with this problem:
https://github.com/dart-lang/pub/issues/3317#issuecomment-1150027103
https://stackoverflow.com/questions/71265574/flutter-pub-publish-returns-error-69-connection-closed-before-full-header-was-r
https://stackoverflow.com/questions/70729747/pub-finished-with-exit-code-69/

---

The problem seems to be particularly common in China due to the Great Firewall. Note that this mirror is recommend in the official Dart/Flutter docs: https://docs.flutter.dev/community/china#configuring-flutter-to-use-a-mirror-site

---

I still don't understand why the Chinese mirror is the one we should be using. The docs seem to imply that it's up to *us* to choose whether we trust this mirror or not.

In any case I don't know of any other mirrors and if it works - it works